scala-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/scala/BaseControlClient.scala

```diff
@@ -0,0 +1,62 @@
+package com.typesafe.conductr.clientlib.scala
+
+import java.net.URL
+
+import com.typesafe.conductr.HttpPayload
+import com.typesafe.conductr.clientlib.scala.models._
+import com.typesafe.conductr.scala.AbstractConnectionContext
+
+import scala.concurrent.Future
+
+private[scala] abstract class BaseControlClient(conductrAddress: URL) {
```

Comment:
Perhaps make this a trait and name it `ControlClientLike`? Having a `Base` and then an `Abstract` subclassing it feels incorrect.